### PR TITLE
APIv2 review corrections #3

### PR DESCRIPTION
**Pull request by baude**
The third pass of corrections for the APIv2.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/runtime_pod.go [hunk 1]
```diff
@@ -182,3 +183,35 @@ func (r *Runtime) GetRunningPods() ([]*Pod, error) {
 	}
 	return runningPods, nil
 }
+
+// PrunePods removes unused pods and their containers from local storage.
+// If force is give, then running pods are also included in the pruning.
```
Comment (mheon) [inline libpod/runtime_pod.go]: Wait, what? I don't think pruning running things makes any sense

File: pkg/api/handlers/containers_create.go [hunk 2]
```diff
@@ -1,16 +1,16 @@
-package generic
+package handlers
 
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/containers/libpod/pkg/api/handlers/generic"
```
Comment (mheon) [inline pkg/api/handlers/containers_create.go]: Import ordering

File: pkg/api/handlers/utils/images.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package utils
 
 import (
 	"fmt"
+	"github.com/gorilla/mux"
```
Comment (mheon) [inline pkg/api/handlers/utils/images.go]: Import ordering

File: pkg/api/server/register_images.go [hunk 4]
```diff
@@ -55,6 +55,27 @@ func (s *APIServer) registerImagesHandlers(r *mux.Router) error {
 	//  - images (compat)
 	// summary: List Images
 	// description: Returns a list of images on the server. Note that it uses a different, smaller representation of an image than inspecting a single image.
+	// parameters:
+	//   - name: "all"
+	//     in: "query"
+	//     description: "Show all images. Only images from a final layer (no children) are shown by default."
```
Comment (mheon) [inline pkg/api/server/register_images.go]: Can't we just say "non-dangling images" instead? It seems much clearer

File: libpod/runtime_pod.go (hunk 1 above)
Comment (TomSweeneyRedHat) [inline libpod/runtime_pod.go]: nit give->given

File: pkg/api/server/register_images.go (hunk 4 above)
Comment (baude) [reply pkg/api/server/register_images.go]: thats what you agreed to earlier ...

File: libpod/runtime_pod.go (hunk 1 above)
Comment (baude) [reply libpod/runtime_pod.go]: @mheon yeah I dont love this ... just because the cli does it does not mean the API must.  Do we agree to remove it?

Comment (mheon) [reply libpod/runtime_pod.go]: Agree. Replace with filters IMO, but we can do that later.

File: pkg/api/handlers/generic/images.go [hunk 5]
```diff
@@ -79,60 +73,23 @@ func PruneImages(w http.ResponseWriter, r *http.Request) {
...
-	//
-	// This code needs to be migrated to libpod to work correctly.  I could not
-	// work my around the information docker needs with the existing prune in libpod.
-	//
-	pruneImages, err := runtime.ImageRuntime().GetPruneImages(!dangling, []image2.ImageFilter{})
+	pruneCids, err := runtime.ImageRuntime().PruneImages(r.Context(), false, filters)
 	if err != nil {
-		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "unable to get images to prune"))
-		return
+		utils.InternalServerError(w, err)
```
Comment (jwhonce) [inline pkg/api/handlers/generic/images.go]: You need a return after this call.

File: pkg/api/handlers/libpod/images.go [hunk 6]
```diff
@@ -110,11 +113,15 @@ func PruneImages(w http.ResponseWriter, r *http.Request) {
 
 	var libpodFilters = []string{}
 	if _, found := r.URL.Query()["filters"]; found {
+		all, err = strconv.ParseBool(query.Filters["all"][0])
+		if err != nil {
+			utils.InternalServerError(w, err)
```
Comment (jwhonce) [inline pkg/api/handlers/libpod/images.go]: You have to return after this Error call

File: pkg/api/handlers/libpod/images.go [hunk 7]
```diff
@@ -171,3 +178,47 @@ func ExportImage(w http.ResponseWriter, r *http.Request) {
...
 	utils.WriteResponse(w, http.StatusOK, rdr)
 }
+
+func ImportImage(w http.ResponseWriter, r *http.Request) {
+	// TODO this is basically wrong
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	runtime := r.Context().Value("runtime").(*libpod.Runtime)
+
+	query := struct {
+		Changes map[string]string `json:"changes"`
```
Comment (jwhonce) [inline pkg/api/handlers/libpod/images.go]: These json schemes should be schema.

File: pkg/api/handlers/libpod/pods.go [hunk 8]
```diff
@@ -326,26 +324,12 @@ func PodPrune(w http.ResponseWriter, r *http.Request) {
...
-		if err != nil {
-			utils.Error(w, "Something went wrong", http.StatusInternalServerError, err)
-			return
-		}
-	} else {
-		// TODO We need to make a libpod.PruneVolumes or this code will be a mess.  Volumes
-		// already does this right.  It will also help clean this code path up with less
-		// conditionals. We do this when we integrate with libpod again.
-		utils.Error(w, "not implemented", http.StatusInternalServerError, errors.New("not implemented"))
+	pruned, err := runtime.PrunePods(query.force)
```
Comment (jwhonce) [inline pkg/api/handlers/libpod/pods.go]: This should be query.Force.  Decoder() will only marshal exported fields.

File: pkg/api/handlers/types.go [hunk 9]
```diff
@@ -537,3 +537,11 @@ func portsToPortSet(input map[string]struct{}) (nat.PortSet, error) {
 	}
 	return ports, nil
 }
+
+// ContainerCreateResponse is the response struct for creating a container
+type ContainerCreateResponse struct {
+	// ID of the container created
+	Id string `json:"Id"`
```
Comment (jwhonce) [inline pkg/api/handlers/types.go]: json fields should be lower case to better match rest of API.

File: pkg/api/server/register_images.go [hunk 10]
```diff
@@ -55,6 +55,27 @@ func (s *APIServer) registerImagesHandlers(r *mux.Router) error {
 	//  - images (compat)
 	// summary: List Images
 	// description: Returns a list of images on the server. Note that it uses a different, smaller representation of an image than inspecting a single image.
+	// parameters:
+	//   - name: "all"
```
Comment (jwhonce) [inline pkg/api/server/register_images.go]: Please remove extra double quotes on map values.

**Comment by baude** [comment]
review comments addressed, please merge

**Pull request merged**